tests/sentry/tasks/test_derive_code_mappings.py

```diff
@@ -21,35 +22,44 @@ def setUp(self):
         )
         self.project = self.create_project(organization=self.organization)
 
+    def generate_data(self, frames: List[Dict[str, Union[str, bool]]], platform: str = ""):
+        test_data = {"platform": platform or self.platform, "stacktrace": {"frames": frames}}
+        return self.store_event(data=test_data, project_id=self.project.id).data
+
 
 class TestJavascriptDeriveCodeMappings(BaseDeriveCodeMappings):
     def setUp(self):
         super().setUp()
-        self.test_data = {
-            "platform": "javascript",
-            "stacktrace": {
-                "frames": [
-                    {
-                        "module": "@sentry/browser/node_modules/@sentry/core/esm/hub",
-                        "filename": "../node_modules/@sentry/browser/node_modules/@sentry/core/esm/hub.js",
-                        "abs_path": "webpack:///../node_modules/@sentry/browser/node_modules/@sentry/core/esm/hub.js",
-                        "in_app": False,
-                    },
-                    {
-                        "module": "app/utils/handleXhrErrorResponse",
-                        "filename": "./app/utils/handleXhrErrorResponse.tsx",
-                        "abs_path": "webpack:///./app/utils/handleXhrErrorResponse.tsx",
-                        "in_app": True,
-                    },
-                ]
-            },
-        }
-        self.event_data = self.store_event(data=self.test_data, project_id=self.project.id).data
+        self.platform = "javascript"
+        self.event_data = self.generate_data(
+            [
+                {
+                    "filename": "../node_modules/@sentry/browser/node_modules/@sentry/core/esm/hub.js",
+                    "in_app": False,
+                },
+                {
+                    "filename": "./app/utils/handleXhrErrorResponse.tsx",
+                    "in_app": True,
+                },
+            ]
+        )
 
     def test_find_stacktrace_paths_single_project(self):
         stacktrace_paths = identify_stacktrace_paths(self.event_data)
         assert stacktrace_paths == ["./app/utils/handleXhrErrorResponse.tsx"]
 
+    def test_find_stacktrace_paths_bad_data(self):
+        data = self.generate_data([{}])
+        data["stacktrace"]["frames"] = [
+            {
+                "abs_path": "https://example.com/static/chunks/foo.bar.js",
+                "data": {"sourcemap": "https://example.com/_next/static/chunks/foo.bar.js.map"},
+                "in_app": True,
+            }
```

Comment:
This represents the values from the Sentry error:
<img width="204" alt="image" src="https://user-images.githubusercontent.com/44410/207391549-ac50b4ce-9e51-40b7-aa0d-fbb75a90a6fc.png">